### PR TITLE
feat(api): File info API 

### LINKED ISSUE #2493: Info
Get info based on ItemId (GET)

### PR DESCRIPTION
**Pull request by soham4abc**
## Description

API endpoint to get info for a file

## How to test
Send a get request to `/file/upload/{UploadId}/item/{ItemId}/info`

## Screenshots
image

closes #2493

@shaheemazmalmmd @GMishx 

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2496"></a>

### CODE REVIEW

**Review by GMishx — CHANGES_REQUESTED** [review]
Seeing the screenshot attached, would recommend following changes:
1. `view_info`:
  1. can extend the `Hash` model object as it already contains the checksums and file size.
  2. No need of different size calculations, the size in bytes stored in `Hash` can be used to calculate them.
  3. Change `filePfileId` to `pfileId`.
2. `meta_info`:
  1. Remove `file` from the beginning of keys.
  2. I believe `fileInfo` and `scanossInfo` as booleans? Please check if so and translate `1 => true`, `0 => false`.

Try uploading this small `deb` file and run the package analysis agent on it to get more info on the page.
At the same time, try running the `scanoss` agent on FOSSology or atarashi to get some match info in Info page.

https://ftp.debian.org/debian/pool/main/l/lsb/lsb-release_11.1.0_all.deb

File: src/www/ui/api/Controllers/FileInfoController.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  private $viewInfo;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->container = $container;
+    $this->clearingDao = $this->container->get('dao.clearing');
+    $this->licenseDao = $this->container->get('dao.license');
+    $this->uploadDao = $container->get('dao.upload');
+    $this->restHelper = $container->get('helper.restHelper');
```
Comment (GMishx) [inline src/www/ui/api/Controllers/FileInfoController.php:63]:
```suggestion
    $this->clearingDao = $this->container->get('dao.clearing');
    $this->licenseDao = $this->container->get('dao.license');
    $this->uploadDao = $this->container->get('dao.upload');
    $this->restHelper = $this->container->get('helper.restHelper');
```

File: src/www/ui/api/Controllers/FileInfoController.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      $response_tag_info = $this->viewInfo->ShowTagInfo($uploadPk, $uploadTreeId);
+      $response_reuse_info = $this->viewInfo->showReuseInfo($uploadPk);
+      $finalValue = array(
+        'view_info' => $response_view,
+        'meta_info' => $response_meta,
+        'package_info' => $response_package_info,
+        'tag_info' => $response_tag_info,
+        'reuse_info' => $response_reuse_info
+      );
+      return $response->withJson($finalValue, 200);
```
Comment (GMishx) [inline src/www/ui/api/Controllers/FileInfoController.php:100]: Please fix indentation.

File: src/www/ui/api/Controllers/FileInfoController.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      $response_tag_info = $this->ViewInfo->ShowTagInfo($uploadPk, $uploadTreeId);
+      $response_reuse_info = $this->ViewInfo->showReuseInfo($uploadPk);
+      $finalValue = array(
+        'view_info' => $response_view,
+        'meta_info' => $response_meta,
+        'package_info' => $response_package_info,
+        'tag_info' => $response_tag_info,
+        'reuse_info' => $response_reuse_info
+      );
+      return $response->withJson($finalValue, 200);
```
[the remark this answers is not in the document]
Comment (GMishx) [reply src/www/ui/api/Controllers/FileInfoController.php:100]: Ummm.. Does not seem to have been resolved.

File: src/www/ui/api/Controllers/FileInfoController.php (hunk 1 above)
Comment (GMishx) [inline src/www/ui/api/Controllers/FileInfoController.php:63]: Setting container is already done in parent class.
```suggestion
    $this->clearingDao = $this->container->get('dao.clearing');
    $this->licenseDao = $this->container->get('dao.license');
    $this->uploadDao = $this->container->get('dao.upload');
    $this->restHelper = $this->container->get('helper.restHelper');
```

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
Tested, working as expected.

**Comment by soham4abc** [comment]
Response for the provided .deb file

Please do have a look at it and we can discuss further changes required!
image

File: src/www/ui/api/Controllers/FileInfoController.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  /**
+   * @var viewInfo $viewInfo
+   * View Info object
+   */
+  private $viewInfo;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->container = $container;
```
Comment (GMishx) [inline src/www/ui/api/Controllers/FileInfoController.php:43]: Parent class initializer does this for you. No need to override it. Please remove the line.

https://github.com/fossology/fossology/blob/3d6195aeff303a01511b5b7662e138f059557f16/src/www/ui/api/Controllers/RestController.php#L52

File: src/www/ui/api/index.php [hunk 5]
```diff
@@ -269,6 +271,13 @@ function (\Slim\Routing\RouteCollectorProxy $app) {
     $app->any('/{params:.*}', BadRequestController::class);
   });
 
+/////////////////////////FILE INFO////////////////////
+$app->group('/file',
+  function (\Slim\Routing\RouteCollectorProxy $app) {
+    $app->get('/upload/{id:\\d+}/item/{itemId:\\d+}/info', FileInfoController::class . ':getItemInfo');
+    $app->any('/{params:.*}', BadRequestController::class);
+  });
+
```
Comment (GMishx) [inline src/www/ui/api/index.php:280]: Since the endpoint is move to `/uploads/{id}/item/{itemId}/info`, shouldn't this be removed as well?

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
> Response for the provided .deb file
> 
> Please do have a look at it and we can discuss further changes required! ![image]

pkgagent does not seem to work correctly now. Let us not bother much about its output as people rarely use it. We can fix it later.

**Comment by soham4abc** [comment]
> > Response for the provided .deb file
> > Please do have a look at it and we can discuss further changes required! ![image]
> 
> pkgagent does not seem to work correctly now. Let us not bother much about its output as people rarely use it. We can fix it later.

Sure!!

File: src/www/ui/api/Controllers/FileInfoController.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+
+class FileInfoController extends RestController
+{
+  /**
+   * @var ContainerInterface $container
+   * Slim container
+   */
+  protected $container;
+
```
Comment (GMishx) [inline src/www/ui/api/Controllers/FileInfoController.php:33]: Unused property. Please remove.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**